### PR TITLE
Switch Pre-provsioned edpm to use IPAM

### PR DESCRIPTION
**Pull request by rabi**
This switches pre-provisioned edpm deployment to use IPAM.

Depends-On: https://github.com/openstack-k8s-operators/dataplane-operator/pull/270

### CODE REVIEW

File: scripts/gen-edpm-kustomize.sh [hunk 1]
```diff
@@ -85,23 +85,25 @@ patches:
       path: /spec/nodes/edpm-compute-0/node/ansibleVars
       value: |
         ctlplane_ip: ${EDPM_COMPUTE_IP}
-        internal_api_ip: 172.17.0.100
-        storage_ip: 172.18.0.100
-        tenant_ip: 172.19.0.100
-        fqdn_internal_api: '{{ ansible_fqdn }}'
-    - op: replace
-      path: /spec/nodes/edpm-compute-0/node/ansibleSSHPrivateKeySecret
-      value: ${EDPM_ANSIBLE_SECRET}
```
Comment (fao89) [inline scripts/gen-edpm-kustomize.sh:94]: why removing the SSH secret?

**Review by fao89 — COMMENTED** [review]
should we update https://github.com/openstack-k8s-operators/dataplane-operator/blob/main/config/samples/dataplane_v1beta1_openstackdataplane.yaml ?

File: scripts/gen-edpm-kustomize.sh (hunk 1 above)
Comment (rabi) [reply scripts/gen-edpm-kustomize.sh:94]: We don't need to duplicate for the nodes when provided for the https://github.com/openstack-k8s-operators/install_yamls/blob/main/scripts/gen-edpm-kustomize.sh#L193

**Review by fao89 — APPROVED** [review]

**Comment by rabi** [comment]
recheck

**Comment by rabi** [comment]
recheck

**Review by stuggi — COMMENTED** [review]
+1 from me for changing the IPs for the nncp here. nice to see we can get rid of all this "hard" coded edpm config.

**Pull request merged**